lib/ansible/plugins/action/__init__.py

```diff
@@ -86,6 +89,15 @@ def run(self, tmp=None, task_vars=None):
                                  ' the tmpdir']
         del tmp
 
+        try:
+            self._admin_users = self._connection._shell.get_option('admin_users')
+            remote_user = self._connection.get_option('remote_user')
+        except AnsibleError:  # old custom plugins w/o get_option
+            self._admin_users = ['root']
+            remote_user = self._play_context.remote_user
```

Comment:
It seems like we need to do each of these separately?  So that if a plugin doesn't have that option we only fallback for that option, not both options?  (And that case will throw KeyError, not AnsibleError iirc).

Reply:
It also doesn't seem right to set these as self values in run().  It seems like we should either set these in __init__() if the action plugin is never reused or pass the values to the functions that use them.